pkg/controller/storagenode/storagenode.go

```diff
@@ -474,10 +479,10 @@ func canNodeServeStorage(storagenode *corev1.StorageNode) bool {
 	// look for node status condition
 	for _, cond := range storagenode.Status.Conditions {
 		if cond.Type == corev1.NodeStateCondition {
-			if cond.Status == corev1.NodeOnlineStatus ||
-				cond.Status == corev1.NodeMaintenanceStatus ||
-				cond.Status == corev1.NodeDegradedStatus {
-				return true
+			if cond.Status != corev1.NodeInitStatus {
+				if ok, err := c.isStorageNode(storagenode, cluster); err == nil {
+					return ok
+				}
 			}
```

Comment:
Previously we were assuming that a node in maintenance state as well as degraded state can serve storage, but that will not happen now right? What will be the impact of this?
Will is be an issue if node goes into maintenance and storage label will be removed from it

Reply:
@svijaykumar-px now we are assuming a node in any state apart from INITIALIZING can serve storage, since storage node can also be in any state.
CC : @vinayakshnd @adityadani @harsh-px @pp511

Reply:
yes @nikita-bhatia we should  consider nodes in maintainence and degreaded state to be part of cluster quorum. If OpenStorage Enumerate node API is returning list of nodes as part of quorum even if those nodes are in maintenance mode or in degraded state then we are good here.

Reply:
That's correct. Thanls for the confirmation @vinayakshnd